syft/source/source.go

```diff
@@ -44,19 +44,17 @@ func New(userInput string, registryOptions *image.RegistryOptions) (*Source, fun
 	case DirectoryScheme:
 		return generateDirectorySource(fs, location)
 	case ImageScheme:
-		return generateImageSource(location, userInput, imageSource, registryOptions)
+		return generateImageSource(location, imageSource, registryOptions)
 	}
 
 	return &Source{}, func() {}, fmt.Errorf("unable to process input for scanning: '%s'", userInput)
 }
 
-func generateImageSource(location, userInput string, imageSource image.Source, registryOptions *image.RegistryOptions) (*Source, func(), error) {
+func generateImageSource(location string, imageSource image.Source, registryOptions *image.RegistryOptions) (*Source,
+	func(), error) {
 	img, err := stereoscope.GetImageFromSource(location, imageSource, registryOptions)
 	if err != nil {
-		log.Debugf("error parsing location: %s after detecting scheme; pulling image: %s", location, userInput)
-		// we may have been to aggressive reading the source hint
-		// try the input as supplied by the user if our initial parse failed
-		img, err = stereoscope.GetImageFromSource(userInput, imageSource, registryOptions)
```

Comment:
I don't think I'm in favor of removing this call to fix the error message since it is catching a lot of cases that might fail initially for the user. 

Could we elevate the `debug` to be an `info` message to give the user context in this scenario that we could not parse the provided location and are trying one more variation?

Reply:
I think we should sync on this. I have a few concerns about not respecting user-specified schemes as explicit intent from the user.